### PR TITLE
Add MIN_TARGET_COVERAGE to HsMetrics

### PR DESCRIPTION
**Pull request by mjhipp**
### Description

The summary statistics provided by HsMetrics are helpful in determining how well a hybrid selection experiment worked, however currently they do not provide an indication if any target(s) achieved lower depth than the mean. While ZERO_CVG_TARGETS can help with this in extreme cases, it does not apply to instances when one target had very low non-zero coverage.

I think MIN_TARGET_COVERAGE is a natural addition to MAX_ and MEAN_ for summarizing a hybrid-selection experiment’s results.

Any thoughts?

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [x] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/analysis/directed/TargetMetricsCollector.java [hunk 1]
```diff
@@ -679,13 +682,15 @@ private void calculateTargetCoverageMetrics() {
                     zeroCoverageTargets++;
                     highQualityCoverageHistogramArray[0] += c.interval.length();
                     targetBases[0] += c.interval.length();
+                    minDepth = 0;
```
Comment (nh13) [inline src/main/java/picard/analysis/directed/TargetMetricsCollector.java]: could you explain this one?

Comment (yfarjoun) [reply src/main/java/picard/analysis/directed/TargetMetricsCollector.java]: I think I understand: there's no coverage `!c.hasCoverage()`....so minDepth=0

File: src/main/java/picard/analysis/directed/TargetMetricsCollector.java [hunk 2]
```diff
@@ -707,6 +712,7 @@ private void calculateTargetCoverageMetrics() {
             metrics.MEAN_TARGET_COVERAGE = (double) totalCoverage / metrics.TARGET_TERRITORY;
             metrics.MEDIAN_TARGET_COVERAGE = highQualityDepthHistogram.getMedian();
             metrics.MAX_TARGET_COVERAGE = maxDepth;
+            metrics.MIN_TARGET_COVERAGE = Math.min(minDepth, maxDepth);
```
[the remark this answers is not in the document]
Comment (yfarjoun) [reply src/main/java/picard/analysis/directed/TargetMetricsCollector.java]: why isn't this just 
```
metrics.MIN_TARGET_COVERAGE = minDepth
```
?

**Comment by yfarjoun** [comment]
My main concern is that this isn't actually useful as I suspect in most experiments there will be a base with no (filtered) coverage, so in the vast majority of cases you'll get MIN_COVERAGE=0 and so it will be a non-informative metric. Have you seen this metric to be useful? may I ask in what experiment types?

File: src/main/java/picard/analysis/directed/TargetMetricsCollector.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjhipp) [reply src/main/java/picard/analysis/directed/TargetMetricsCollector.java]: This is to handle an edge case where the target interval list is empty. 
In that case, minDepth = Long.MAX_VALUE, and maxDepth=0.

File: src/main/java/picard/analysis/directed/TargetMetricsCollector.java (hunk 1 above)
Comment (mjhipp) [reply src/main/java/picard/analysis/directed/TargetMetricsCollector.java]: Yes, in the case where any target interval has no coverage, the minDepth must be 0

**Comment by mjhipp** [comment]
> My main concern is that this isn't actually useful as I suspect in most experiments there will be a base with no (filtered) coverage, so in the vast majority of cases you'll get MIN_COVERAGE=0 and so it will be a non-informative metric. Have you seen this metric to be useful? may I ask in what experiment types?

When designing baits (for targeted hybrid-selection capture experiments) to completely cover all target intervals, it is very rare in my experience to have 0 depth at any base in the target intervals (even after filtering and error correction steps) unless there was an extreme failure of equipment or user error. However there is quite often a case when a pool of baits used in these experiments is unbalanced, or a subset of baits should have their concentrations increased within a pool. Being able to compare MAX, MEAN, as well as MIN target coverage is a fast way to tell if more attention needs to be paid when mixing baits for experiments in order to obtain uniform target coverage.

**Comment by mjhipp** [comment]
@pshapiro4broad Any chance for a review? I would like to add this feature to our pipeline :)

**Comment by mjhipp** [comment]
Thanks, comment added (a11cd34)

**Pull request merged**